### ISSUE TITLE
[Bug]: Postgres package runs (and fails) as a service.

### ISSUE THREAD

**Issue by Izhaki** [jetpack-io/devbox#1026]
We have started trailing `devbox` and it is awesome, borderline life-changing.

One niggling issue that we couldn't find an answer to in the docs.

Our environment requires `postgresql` - we need its cli. So we have added it to the `packages` field in `devbox.json`. No `includes`.

Our actual postgres DB is dockerised, and declared as a service in `process-compose.yml`.

Whenever we run `devbox services up`, we get both an instance of `postgresql`, and our own docker one. The `postgresql` service just spits tons of errors - `.../devbox/v-6cee4/postgresql/data is not a database cluster directory`.

Ideally, when install under `packages` postgres should not run as a service.

**Comment by loreto** [comment]
@Izhaki Thanks for reporting. Just to make sure I'm understanding correctly: you are wanting to install the `postgresql` package, _without_ starting it as a service, since you already do that separately. Instead of packages automatically starting services, you want to distinguish between two cases:
1. Install this package
2. Install this package and also start it's service.

And that way you can pick which of the two cases you want.

Is that roughly right?

**Comment by Lagoja** [comment]
Since you're using `devbox services up`, I'm assuming that there are other services that you _do_ want Devbox to start. So I think for this to work, we'd need a way to tell Devbox not to include the service when activating the plugin, or to skip it when running `devbox services up`. We could add something to the config or the includes to make this possible. 

@Izhaki  There are also two not-ideal workarounds that could unblock you for now:

1. In your `process-compose.yml`, you could rename the Dockerized postgres process to `postgresql`. That should overwrite the plugin service in process-compose, and only launch your Dockerized version

2. If you import postgresql as a flake reference, it won't activate the plugin. You can do this by replacing `postgresql` in your packages with 

```
github:nixos/nixpkgs/nixpkgs-unstable#postgresql
```

**Comment by Izhaki** [comment]
> 1. Install this package
> 2. Install this package and also start it's service.
> And that way you can pick which of the two cases you want.

Exactly. Said another way, we want to be able to install the package, but not have it running as a service (1)... (2) is what happens right now.

> github:nixos/nixpkgs/nixpkgs-unstable#postgresql

Thanks! That did the job!

---

Proposal.

The [Using Plugins](https://www.jetpack.io/devbox/docs/guides/plugins/#using-plugins) says:

> If you add one of the packages listed above to your project using `devbox add <pkg>`, Devbox will automatically activate the plugin for that package.

And then says you can also do this:

```json
{
  "includes": [
    "plugin:nginx"
  ]
}
```

Possibly `devbox add` means - don't use the plugin (and hence no service when `services up`), whereas `includes` means "I want this as a plugin, including the services that come along it".

I know this is a breaking change, but seems like a reasonable option to me.

**Comment by yannick-mayeur** [comment]
I've been wanting to do the same think. On my end I solved it like @Lagoja mentionned renaming my dockerized postgresql service to override the plugin one.

The use case is that I need postgresql as a package for rails, but I don't want it to handle my database (at least not yet as I want to increasingly adopt devbox).

I agree with @Izhaki it would be nice for the plugins to be added explicitly in the configuration. Another solution that is not a breaking change would be to explicitly be able to exclude some plugins with somethink like this:
```
 {
  "excludes": [
    "plugin:nginx"
  ]
}
```
